internal/plugincommon/consensus/threshold.go

```diff
@@ -0,0 +1,47 @@
+package consensus
+
+// Threshold is the minimum number of observations required for a given chain.
+type Threshold uint
+
+// MultiThreshold is an interface that maps a chain selector to a threshold.
+type MultiThreshold[T any] interface {
+	// Get the minimum number of observations required for a given chain.
+	Get(chain T) (Threshold, bool)
```

Comment:
If this doesn't have to be a chain, maybe we can call it "minimum number of observations required for a given scope"? Not sure what's a good name for this.

Reply:
I went with `key`